src/settings.cpp

```diff
@@ -1621,7 +1751,7 @@ StringList GetGRFPresetList()
 {
 	StringList list;
 
-	std::unique_ptr<IniFile> ini(IniLoadConfig());
+	std::unique_ptr<IniFile> ini = std::make_unique<ConfigIniFile>(_config_file);
```

Comment:
You could also do without the `std::unique_ptr` completely. It would then be like `LoadFromConfig`.